### PR TITLE
newsletter-toggle-clicked event

### LINKED ISSUE #47280: Analytics event for newsletter subscription opt-in/opt-out during the setup

### PR DESCRIPTION
**Pull request by npretto**
Closes https://github.com/metabase/metabase/issues/47280

It uses the new generic event schema to track opt-in/opt-out from the newsletter in the setup

### CODE REVIEW

File: frontend/src/metabase-types/analytics/event.ts [hunk 1]
```diff
@@ -17,4 +17,10 @@ export type CsvUploadClickedEvent = ValidateEvent<{
   triggered_from: "left_nav";
 }>;
 
-export type SimpleEvent = CsvUploadClickedEvent;
+export type NewsletterToggleClickedEvent = {
```
Comment (ranquild) [inline frontend/src/metabase-types/analytics/event.ts:20]: @npretto you'd need to call `ValidateEvent` like for `CsvUploadClickedEvent`. You'll see issues immediately.

Comment (npretto) [reply frontend/src/metabase-types/analytics/event.ts]: Yeah I rebased what I had quickly without double checking, my bad

**Review by ranquild — APPROVED** [review]
I believe this was easier than before :)

**Comment by ranquild** [comment]
@npretto I'll deal with backporting the base PR today, just FY.

**Review by heypoom — APPROVED** [review]
The event looks good!

**Review by Somtom — APPROVED** [review]

**Pull request merged**